pkg/apis/v1/nodepool.go

```diff
@@ -56,6 +57,17 @@ type NodePoolSpec struct {
 	// +kubebuilder:validation:Maximum:=100
 	// +optional
 	Weight *int32 `json:"weight,omitempty"`
+	// Replicas is the desired number of nodes for the NodePool. When specified, the NodePool will
+	// maintain this fixed number of replicas rather than scaling based on pod demand.
+	// When replicas is set, the following fields are simply ignored:
+	// - disruption.consolidationPolicy
+	// - disruption.consolidateAfter
+	// - limits.cpu
+	// - limits.memory
```

Comment:
Had some comments regarding this in the RFC

Reply:
Addressed here - https://github.com/kubernetes-sigs/karpenter/pull/2309#discussion_r2256017180